### ISSUE TITLE
[Bug] How to force the closure?

### ISSUE THREAD

**Issue by marciusnery** [Blazored/Modal#117]
**Describe the bug**
I am opening a modal to display a "waiting" image while processing is done. At the conclusion of the process I am calling the BlazoredModal.Close() method but it is not closing.

**To Reproduce**
```
public async Task ReadFile()
    {
        Modal.Show<ModalMessage>("", ModalConfig.GetParameters("Waiting.", "", "", false, "Loading"));
        Message = string.Empty;
        this.StateHasChanged();

        file = (await Reader.EnumerateFilesAsync()).FirstOrDefault();
        if (file != null)
        {
            try
            {
                Message = "Reading file...";
                this.StateHasChanged();
                var fileInRam = await file.CreateMemoryStreamAsync(4096);
                Message = "Parsing file...";
                this.StateHasChanged();
                using (var package = new OfficeOpenXml.ExcelPackage(fileInRam))
                {
                    Tables = EPPlusBuilder.ParseByWorksheetIndex(package, FirstRowIsHeader, 1);
                }
                this.StateHasChanged();
                Message = "Done.";
                ArquivoCarregado = true;                
                filter.Recebimentos = new List<RecCreditoCip>();
                filter.Recebimentos = await ParseTable.ToRecCreditoCipListAsync(Tables);
                this.StateHasChanged();
                BlazoredModal.Close();
            }
            catch (Exception e)
            {
                Message = e.ToString();
            }
        }
    }
```

**Hosting Model (is this issue happening with a certain hosting model?):**
 - Blazor WebAssembly

**Comment by chrissainty** [comment]
The code you’ve included, is that being run from the first Modal instance?

**Comment by marciusnery** [comment]
The complete code:

`<button type="button" @onclick=ConfirmReading class="btn btn-primary btn-block">Open File</button>`

The first modal asks for confirmation if the user wants to continue.

```
protected async void ConfirmReading()
    {        
            StateHasChanged();
            var modalRef = Modal.Show<ModalMessage>("", ModalConfig.GetParameters("Confirm?", "", "Fechar", true, "Question"));
            var result = await modalRef.Result;
            if (!result.Cancelled)
            {
                await ReadFile();                
            }        
    }
```

```
public async Task ReadFile()
    {
        Modal.Show<ModalMessage>("", ModalConfig.GetParameters("Waiting.", "", "", false, "Loading"));
        Message = string.Empty;
        this.StateHasChanged();

        file = (await Reader.EnumerateFilesAsync()).FirstOrDefault();
        if (file != null)
        {
            try
            {
                Message = "Reading file...";
                this.StateHasChanged();
                var fileInRam = await file.CreateMemoryStreamAsync(4096);
                Message = "Parsing file...";
                this.StateHasChanged();
                using (var package = new OfficeOpenXml.ExcelPackage(fileInRam))
                {
                    Tables = EPPlusBuilder.ParseByWorksheetIndex(package, FirstRowIsHeader, 1);
                }

                this.StateHasChanged();
                Message = "Done.";
                ArquivoCarregado = true;
                filter.Recebimentos = new List<RecCreditoCip>();
                filter.Recebimentos = await ParseTable.ToRecCreditoCipListAsync(Tables);
                this.StateHasChanged();
                BlazoredModal.Close();
            }
            catch (Exception e)
            {
                Message = e.ToString();
            }
        }
    }
```

**Comment by chrissainty** [comment]
How are you closing the second Modal? From the looks of your code, the second Modal is still active?

**Comment by marciusnery** [comment]
I want to know how do I close. The Modal.Close() method no longer exists.

**Comment by chrissainty** [comment]
You can only close a Modal instance from that instance. You can’t close it remotely.

**Comment by marciusnery** [comment]
In this case how do I close?

Please say you have a solution. LOL

**Comment by chrissainty** [comment]
As I said, you can’t close Modal instances remotely. 

I would suggest reworking your implementation. For example, in your first modal, when the user confirms the action, you could trigger your long running task and update the UI there.

**Comment by marciusnery** [comment]
Here is my first modal. I wait for the result of it to call the processing.

```
protected async void ConfirmReading()
    {        
            StateHasChanged();
            var modalRef = Modal.Show<ModalMessage>("", ModalConfig.GetParameters("Confirm?", "", "Fechar", true, "Question"));
            var result = await modalRef.Result;
            if (!result.Cancelled)
            {
                await ReadFile();                
            }        
    }
```

Could you give an example of how to solve this problem?

**Comment by benhysell** [comment]
I just ran into this issue as well...

Was hoping to:

- Throw up a modal with a spinner and text, i.e. "Submitting"
- Do the actual work of sending the data to the server, checking for success or fail, etc.
- On success or fail take down the modal programmatically and throw up a toast with the resulting status

Am I reading this thread correctly...it sounds like once the modal is up the modal is out of my hands and I can't interact with it from where I called it from?  i.e. I can't force a close from where it was called?
